.github/README.md

```diff
@@ -0,0 +1,24 @@
+This README explains the CI design and how you can update the CI workflow.
+
+## Overall Design
+The CI is consisted with two parts: 
+1. GitHub Action
+    * Push and Pull Request events will trigger GitHub Action workflow defined in `.github/workflows/continuous_integratoin.yml`.
+    * It will then checkout code on **master branch of autogluon** if it's a Pull Request event, or checkout code on **corresponding branch of autogluon** if it's a Push event, to fetch the `submit-job.py` script, which will kick off AWS Batch job to run tests/build docs, under `CI/batch`.
+2. AWS Batch.
+    * AWS Batch job runs in container defined under `CI/batch/docker`.
+    For details on how to build and push the docker image, please refer to `CI/batch/docker/README.md`
+    * It will execute scripts defined under `.github/worfklow_scripts` for various tasks.
+    It will only use scripts updated by Pull Request if the author has write permission to our repo. Otherwise, scripts from **master branch of autogluon** will be used.
+
+## How to update the workflows
+
+### Regular Contributors
+**IMPORTANT:** You are only able to update the workflow if you have **write permission** to our repo for security concern. You are still able to add/modify unit tests/ docs and see the changes. It's just you cannot modify the workflow.
+
+### Maintainers
+For maintainers with write permission, the easiest way to update our workflow is push to a branch under `awslabs/autogluon`.
+The Push event will trigger the workflow to reflect your latest changes.
+Once you are satisfied with the changes and the CI passed, start a pull request. CI for both Push and Pull Request event will be triggered in the pull request, and **only the one with Push event reflects your latest workflow changes**
```

Comment:
Yes. So if you push to a branch of our autogluon repo, it will trigger both a Push workflow and a Pull Request workflow. The Push workflow will actually checkout your branch while the Pull Request workflow will checkout out master.

On the other hand, if you start PR on a fork. Only workflow from the master will be checked out. I can improve the readme a little bit :).